### PR TITLE
x shaped crosshair

### PR DESCRIPTION
**Pull request by newcommerdontblame**
meme feature that i think pretty cool with noscope awp
(recommend to use in rainbow mode speed 40=))))
image
ss on 800x640 so dont mind quality u can configure lines if u want to add outline or something..hf with it

### CODE REVIEW

**Comment by dayz2000** [comment]
best feature!!!!

File: Osiris/Hacks/Visuals.cpp [hunk 1]
```diff
@@ -406,3 +406,32 @@ void Visuals::skybox(FrameStage stage) noexcept
         memory->loadSky(sv_skyname->string);
     }
 }
+
+void Visuals::xCrosshair() noexcept
+{
+    static auto crosshairalpha = interfaces->cvar->findVar("cl_crosshairalpha");
```
Comment (Khajiitos) [inline Osiris/Hacks/Visuals.cpp:412]: Maybe some people use a crosshair with less than 255 alpha, why don't you use the cvar "crosshair" instead?

File: Osiris/Hacks/Visuals.cpp [hunk 2]
```diff
@@ -406,3 +406,32 @@ void Visuals::skybox(FrameStage stage) noexcept
...
+    crosshairalpha->setValue(255);
+
+    if (!config->visuals.xCrosshair.enabled)
+        return;
+
+    crosshairalpha->setValue(0);
+
+    if (!interfaces->engine->isInGame() || !localPlayer.get()->isAlive())
```
Comment (Khajiitos) [inline Osiris/Hacks/Visuals.cpp:420]: localPlayer is already an entity, no need to use .get()

File: Osiris/Hacks/Visuals.cpp [hunk 3]
```diff
@@ -406,3 +406,32 @@ void Visuals::skybox(FrameStage stage) noexcept
...
+        return;
+    
+        const auto [width, height] = interfaces->surface->getScreenSize();
+
+        const auto width_mid = width / 2;
+        const auto height_mid = height / 2;
+ 
+        if (config->visuals.xCrosshair.rainbow)
```
Comment (Khajiitos) [inline Osiris/Hacks/Visuals.cpp:428]: so if crosshair rainbow is enabled, then set the draw color to rainbow but not draw the crosshair?

File: Osiris/Hacks/Visuals.cpp [hunk 4]
```diff
@@ -406,3 +406,32 @@ void Visuals::skybox(FrameStage stage) noexcept
...
+            interfaces->surface->setDrawColor(rainbowColor(config->visuals.xCrosshair.rainbowSpeed));
+        else
+            interfaces->surface->setDrawColor(config->visuals.xCrosshair.color);
+            interfaces->surface->drawLine(width_mid + 10, height_mid + 10, width_mid + 4, height_mid + 4);
+            interfaces->surface->drawLine(width_mid - 10, height_mid + 10, width_mid - 4, height_mid + 4);
+            interfaces->surface->drawLine(width_mid + 10, height_mid - 10, width_mid + 4, height_mid - 4);
+            interfaces->surface->drawLine(width_mid - 10, height_mid - 10, width_mid - 4, height_mid - 4);
+
```
Comment (Khajiitos) [inline Osiris/Hacks/Visuals.cpp:436]: Shouldn't you use brackets here?

File: Osiris/Hacks/Visuals.cpp (hunk 3 above)
Comment (newcommerdontblame) [reply Osiris/Hacks/Visuals.cpp:428]: wdym it draws

File: Osiris/Hacks/Visuals.cpp (hunk 1 above)
Comment (newcommerdontblame) [reply Osiris/Hacks/Visuals.cpp:412]: 10000in1 lol

Comment (newcommerdontblame) [reply Osiris/Hacks/Visuals.cpp:412]: thats 1 in 10000 situation lol u can configure the code too. im not saying mine is clearest one 
just 5 minute stuff cuz someone asked

**Pull request closed without merging**